conf.d/abbr_tips.fish

```diff
@@ -31,6 +31,34 @@ function __abbr_tips --on-event fish_postexec -d "Abbreviation reminder for the
                 set -e __ABBR_TIPS_KEYS[$abb]
                 set -e __ABBR_TIPS_VALUES[$abb]
         end
+    else if test "$command[1]" = "alias"
+        # Update abbreviations list when adding aliases
+        set -l alias_key
+        set -l alias_value
+
+        if string match -q '*=*' "$command[2]"
```

Comment:
Since `alias foo=bar baz` will create a tip which does not match with the alias actually created (tip `foo -> bar baz` for the alias `foo=bar -> baz`), we can prevent adding the tip if the length of `command_split[2]` is greater than 1 (when the aliased command is more than one word).

Reply:
I'm not sure that logic is quite right.

Because `command` is split on spaces, `command_split[2]` will only ever be the first word of the alias and `command_split` will never have more than two parts.

The two cases seem like they're indistinguishable:

```
❯ alias foo=ls bar
$command_split[1] foo
$command_split[2] ls
$command[1] alias
$command[2] foo=ls
$command[3] bar
```

```
❯ alias foo="ls bar"
$command_split[1] foo
$command_split[2] "ls
$command[1] alias
$command[2] foo="ls
$command[3] bar"
```

The first is invalid and the second is valid, but the number of items is identical. And then there's actually a third case, also valid:

```
❯ alias foo=ls\ bar
$command_split[1] foo
$command_split[2] ls\
$command[1] alias
$command[2] foo=ls\
$command[3] bar
```

One option would be to check that the `=` is followed by `'` or `"`, or that `command[2]` ends with `\`, but this begins to be very messy logic for such a small case.

I think the only solution would be to ignore every alias defined with an `=` where `count $command` is greater than 2. This will exclude some valid aliases, but users can always use the syntax without `=` instead. (I think the `=` form is probably for compatibility with POSIX shell.)

Reply:
You get it right. It's safe to limit the `alias a=b` way since the other options still works for all aliases anyway.